### PR TITLE
[FIX] account: fix update taxes link between parent and children tax

### PR DESCRIPTION
**Pull request by clbr-odoo**
When updating taxes, all taxes are correctly created but the link between children taxes and their parent was not correctly set.

opw-3347425 (1st issue)

### CODE REVIEW

File: addons/account/tests/test_chart_template.py [hunk 1]
```diff
@@ -361,3 +375,23 @@ def test_update_taxes_fiscal_country_check(self):
...
+
+        parent_tax = self.env['account.tax'].search([
+            ('company_id', '=', self.company.id),
+            ('name', '=', group_tax_name),
+        ])
+        children_taxes = self.env['account.tax'].search([
+            ('company_id', '=', self.company.id),
+            ('name', 'like', f'{group_tax_name}_%'),
+        ])
+        self.assertEqual(len(parent_tax), 1, "The parent tax should've been created.")
```
Comment (oco-odoo) [inline addons/account/tests/test_chart_template.py:395]: should have *

File: addons/account/tests/test_chart_template.py [hunk 2]
```diff
@@ -361,3 +375,23 @@ def test_update_taxes_fiscal_country_check(self):
...
+        self.assertEqual(len(children_taxes), 2, "Two children should've been created.")
```
Comment (oco-odoo) [inline addons/account/tests/test_chart_template.py:396]: should have *

File: addons/account/models/chart_template.py [hunk 3]
```diff
@@ -65,7 +65,7 @@ def _avoid_name_conflict(company, template):
                 for index, conflict_taxes in enumerate(conflict_taxes):
                     conflict_taxes.name = f"[old{index if index > 0 else ''}] {conflict_taxes.name}"
 
-        templates_to_create = env['account.tax.template']
+        templates_to_create = env['account.tax.template'].with_context(active_test=False)
```
Comment (oco-odoo) [inline addons/account/models/chart_template.py:68]: Not really related to what you're fixing, is it ? You're basically making sure inactive templates also get updated (which is correct); shouldn't it be done in a distinct commit, with a dedicated test ?

Comment (clbr-odoo) [reply addons/account/models/chart_template.py:68]: It is actually: inactive templates were added to the recordset, only their `children_tax_ids` were ignored.
But I agree I should add tests for inactive children and templates.

**Comment by oco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**